operationsgateway_api/src/routes/filters.py

```diff
@@ -0,0 +1,176 @@
+from http import HTTPStatus
+import logging
+
+from fastapi import APIRouter, Depends, Path, Query, Response, status
+from fastapi.responses import JSONResponse
+from typing_extensions import Annotated
+
+from operationsgateway_api.src.auth.authorisation import authorise_token
+from operationsgateway_api.src.auth.jwt_handler import JwtHandler
+from operationsgateway_api.src.error_handling import endpoint_error_handling
+from operationsgateway_api.src.users.favourite_filter import FavouriteFilter
+
+
+log = logging.getLogger()
+router = APIRouter()
+AuthoriseToken = Annotated[str, Depends(authorise_token)]
+
+
+@router.get(
+    "/users/filters",
+    summary="Get a list of the user's favourite filters",
+    response_description="List of favourite filters",
+    tags=["Filters"],
+)
+@endpoint_error_handling
+async def get_favourite_filters(access_token: AuthoriseToken):
+    """
+    Get a list of the user's favourite filters that belong to the user sending the
+    request
+    """
+
+    token_payload = JwtHandler.get_payload(access_token)
+    username = token_payload["username"]
+    log.info("Getting favourite filters for user %s", username)
+
+    return await FavouriteFilter.get_list(username)
+
+
+@router.get(
+    "/users/filters/{_id}",
+    summary="Get a list of the user's favourite filters",
+    response_description="List of favourite filters",
+    tags=["Filters"],
+)
+@endpoint_error_handling
+async def get_favourite_filter_by_id(
+    _id: Annotated[
+        str,
+        Path(
+            ...,
+            description="`_id` of the filter to remove from the database",
+        ),
+    ],
+    access_token: AuthoriseToken,
+):
+    """
+    Get a specific favourite filter given its ID
+    """
+
+    token_payload = JwtHandler.get_payload(access_token)
+    username = token_payload["username"]
+    log.info("Getting favourite filter of _id '%s' for user '%s'", _id, username)
+    return await FavouriteFilter.get_by_id(username, _id)
+
+
+@router.post(
+    "/users/filters",
+    summary="Store a favourite filter into the user's document in the database",
+    response_description="List of favourite filters",
+    tags=["Filters"],
+)
+@endpoint_error_handling
+async def create_favourite_filter(
+    access_token: AuthoriseToken,
+    name: str = Query(
+        "",
```

Comment:
Should `name` be optional with a default of `""`? Is it meaningful to create a favourite filter without a name (or should it always be required)?

Reply:
Good point, this endpoint is intended to have both query parameters be mandatory. I've fixed this in 1c493c6e455dcbd9a900d7eb689c5381bc6fc603